hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/HoodieSparkUtils.scala

```diff
@@ -107,23 +107,19 @@ object HoodieSparkUtils extends SparkAdapterSupport with SparkVersionsSupport wi
     //       injecting [[SQLConf]], which by default isn't propagated by Spark to the executor(s).
     //       [[SQLConf]] is required by [[AvroSerializer]]
     injectSQLConf(df.queryExecution.toRdd.mapPartitions { rows =>
-      if (rows.isEmpty) {
-        Iterator.empty
```

Comment:
Does removal of this provide any benefit?

Reply:
Yes, this will trigger the dag to see if it is in fact returning an empty set of rows. You can see this on the spark UI when running your jobs.

Reply:
I was looking at the wrong place. This is the spot that needed to be updated: https://github.com/apache/hudi/pull/10435/files#diff-21ccee08cebace9de801e104f356bf4333c017d5d5c0cac4e3de63c7861f3c13R100

Reply:
Got it.